sklearnex/linear_model/ridge.py

```diff
@@ -236,18 +231,21 @@ def _onedal_gpu_supported(self, method_name, *data):
             )
 
             if method_name == "fit":
-                n_samples = _num_samples(data[0])
-                n_features = _num_features(data[0], fallback_1d=True)
-                is_underdetermined = n_samples < (n_features + int(self.fit_intercept))
-                patching_status.and_conditions(
-                    [
-                        (
-                            not is_underdetermined,
-                            "The shape of X (fitting) does not satisfy oneDAL requirements:"
-                            "Number of features + 1 >= number of samples.",
-                        ),
-                    ]
-                )
+                if isinstance(self.alpha, numbers.Real) and np.isclose(self.alpha, 0):

```

Comment:
I don't think `np.isclose` would suffice here. This regularizaation number is meant to be added to a large number which is the result of sums of squares. Therefore, at the very least, it should be larger than machine epsilon for the desired dtype, and likely higher in order to ensure that $ssq + \alpha > ssq$.

Perhaps 1e-5 could be a more reasonable minimum threshold.